### DISCUSSION TITLE
Expected plan to launch Borg2 as stable

### DISCUSSION THREAD

**Discussion by lamyergeier** [borgbackup/borg#8571] [Q&A] [answered]
@ThomasWaldmann  Which of the following is a likely scenario?

1 months from 01.12.2024
3 months from 01.12.2024
6 months from 01.12.2024
1 year from 01.12.2024
longer

**Comment by ThomasWaldmann** [comment, marked as answer]
Well, there is no plan, it will be released when ready.

How long that takes will depend on:
- amount of bugs / issues found in beta test.
- amount of contributors helping with testing, docs, fixing issues, implementing stuff, etc.
- whether "one storage object per chunk" will be kept as it currently is or "pack files / segment files / index" will come back before a release is made.

That said, I guess I can say that it for sure won't be in Q1/2025.

You can check the milestones on the github issue tracker to get a rough idea about progress, but even with that it is hard to say.

**Reply by akostadinov** [reply]
Hello. Could you clarify which milestone needs to be tracked and which would be the blocker issues for the stable release? I see a lot of issues for milestone 2.x, 2beta and 2cr1. Is there a list of blockers?

And another question - are there any other active contributors except you? I see commits to master only from you. I hope some people at least donate. I don't have time to directly contribute but this is some serious work that I hope is sustainable.

**Reply by ThomasWaldmann** [reply]
One of the main big things still todo is #8572.

There are quite some people contributing (docs, testing, user support, donations [although maybe only a low percentage of all users donating]), but most of the coding work is currently done by me.

Besides "having time", there is also the issue of few "exciting, low hanging fruit issues".

Either issues are a bit boring (like updating the docs) or they are rather complex. That was easier in the early borg days when there were a lot of bugs to fix and a lot of smaller features to add.

Also, some stuff in borg 1.x code is rather complex and hard to work with. I try to make borg2 simpler and still better than borg1, but that is often a breaking change. I also try to put all breaking changes into borg2, because breaking changes cause work for users.
